File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/GetPropertyDescriptorRequest.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.jspecify.annotations.Nullable;
import org.opcfoundation.webserver.types.digitaltwin.ObjectServiceContext;

public class GetPropertyDescriptorRequest {
    private final String id;
    private final String propertyId;
    private final @Nullable String subElementName;
    private final ObjectServiceContext context;

    public GetPropertyDescriptorRequest(
            ObjectServiceContext context,
            String propertyId,
            @Nullable String subElementName)
    {
        this.id = context.getObjectId().getId();
        this.propertyId = propertyId;
        this.subElementName = subElementName;
        this.context = context;
    }

    public String getId() {
        return id;
    }

    public String getPropertyId() {
        return propertyId;
    }

    public @Nullable String getSubElementName() {
        return subElementName;
    }

    public ObjectServiceContext getContext() {
        return context;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/GetPropertyDescriptorRequest.ts
import { ObjectServiceContext } from "../../../types/digitaltwin/ObjectServiceContext";

export class GetPropertyDescriptorRequest {
    private readonly _id: string;
    private readonly _propertyId: string;
    private readonly _subElementName: string | null;
    private readonly _context: ObjectServiceContext;

    constructor(
        context: ObjectServiceContext,
        propertyId: string,
        subElementName: string | null = null) {
        this._id = context.objectId.id;
        this._propertyId = propertyId;
        this._subElementName = subElementName;
        this._context = context;
    }

    get id(): string {
        return this._id;
    }

    get propertyId(): string {
        return this._propertyId;
    }

    get subElementName(): string | null {
        return this._subElementName;
    }

    get context(): ObjectServiceContext {
        return this._context;
    }
}